src/ports/user/model.ts

```diff
@@ -0,0 +1,28 @@
+import type { Address, ID, Key, HashPassword, String, Boolean } from '../types';
+
+export enum ROLE {
+  /* Full permissions to any resources (app, database...) */
+  ADMIN = 'ADMIN',
+  /* Grants write and read permissions on their own resources and `CLIENT` resources without `ADMIN` permissions */
+  EDITOR = 'EDITOR',
+  /* Grants write and read permissions on their own resources  */
+  CLIENT = 'CLIENT',
+}
```

Comment:
personally think enums are not very good in typescript [^1]

[^1]: https://graphqleditor.com/blog/enums-are-still-bad/